File: Practice Arrays/11Intersecting arrays.py
def intersect(nums1, nums2):
    # Write your code here
    
    ct1={}
    ct2={}
    for num in nums1:
        if num not in ct1:
            ct1[num]=1
        else:
            ct1[num]+=1
    for num in nums2:
        if num not in ct2:
            ct2[num]=1
        else:
            ct2[num]+=1
    res=[]
    for i in range(101):
        if i in ct1 and i in ct2:
            times=min(ct1[i],ct2[i])
            res.extend([i]*times)
    return res
    

def main():
    num1size, num2size = list(map(int, input().split()))
    nums1 = list(map(int, input().split()))
    nums2 = list(map(int, input().split()))
    ans = intersect(nums1, nums2)
    print(*ans)

if __name__ == "__main__":
    main()

File: Practice Arrays/codigo_cs/11Intersecting arrays.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Test11
{

    static List<int> Intersect(int[] nums1, int[] nums2)
    {
        Dictionary<int, int> ct1 = new Dictionary<int, int>();
        Dictionary<int, int> ct2 = new Dictionary<int, int>();

        foreach (int num in nums1)
        {
            if (ct1.TryGetValue(num, out int count))
            {
                ct1[num] = count + 1;
            }
            else
            {
                ct1[num] = 1;
            }
        }

        foreach (int num in nums2)
        {
            if (ct2.TryGetValue(num, out int count))
                ct2[num] = count + 1;
            else
                ct2[num] = 1;
        }

        List<int> res = new List<int>();

        for (int i = 0; i <= 100; i++)
        {
            if (ct1.TryGetValue(i, out int value1) && ct2.TryGetValue(i, out int value2))
            {
                int times = Math.Min(value1, value2);
                for (int j = 0; j < times; j++)
                {
                    res.Add(i);
                }
            }
        }

        return res;
    }
    public static void IntersectingArrays()
    {
        int[] sizes = Console.ReadLine().Split().Select(int.Parse).ToArray();
        int[] nums1 = Console.ReadLine().Split().Select(int.Parse).ToArray();
        int[] nums2 = Console.ReadLine().Split().Select(int.Parse).ToArray();

        List<int> ans = Intersect(nums1, nums2);
        Console.WriteLine(string.Join(" ", ans));

    }
}
